src/main/java/me/moodcat/database/controllers/ChatDAO.java

```diff
@@ -0,0 +1,26 @@
+package me.moodcat.database.controllers;
+
+import static me.moodcat.database.entities.QChatMessage.chatMessage;
+
+import java.util.List;
+
+import javax.persistence.EntityManager;
+
+import me.moodcat.database.entities.ChatMessage;
+import me.moodcat.database.entities.Room;
+
+public class ChatDAO extends AbstractDAO<ChatMessage> {
+
+    private static final long NUMBER_OF_CHAT_MESSAGE = 10;
+
+    protected ChatDAO(final EntityManager entityManager) {
+        super(entityManager);
+    }
+
+    public List<ChatMessage> listByRoomId(final Room roomId) {
```

Comment:
I think it is wise to keep a log of chat messages for analyzing-purposes. We can implement a cache for the retrieval of messages and when a message is posted both cache and log it.